### ISSUE TITLE
Initial running of TensorFlow on WSL Ubuntu 20.04 looks very very slow.

### ISSUE THREAD

**Issue by jinzzasol** [tensorflow/tensorflow#55067]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **WSL2 Ubuntu 20.04**
- TensorFlow version (use command below)**: 2.1.0**
- Python version: **Python 3.7**
- CUDA/cuDNN version:
       **nvcc: NVIDIA (R) Cuda compiler driver
       Copyright (c) 2005-2019 NVIDIA Corporation
       Built on Sun_Jul_28_19:07:16_PDT_2019
       Cuda compilation tools, release 10.1, V10.1.243**
- GPU model and memory: **RTX 3050 Ti Laptop GPU**

# Issue

Dear all,

I would like to ask for help to solve the issue I've been experiencing with Tensorflow on WSL2.
I'm a relatively light user and a novice of Tensorflow and I've been using different machine learning (ML) models on Windows. Recently, I found that I can take some advantages of performing MLs using Linux on WSL2 instead of Windows, so I set the environment in WSL2 and was able to run the code with GPU. (It was a very painful process to set the same environment as it is on Windows.) 

However, when I run my code in WSL Ubuntu, I found that the **(1) initial running takes very very long compare to Windows.** Also, **(2) there are multiple lines showing which I didn't see with Windows. And it only appears in the initial run.** You can find them in the images below.
Since the code is running, I believe that the code is using GPU properly on WSL2 system. But if it takes this long, there is no reason for using Linux on WSL2 instead of Windows. Maybe this is true since I'm a light user, but I want to learn a Linux and develop my capability. So I'm asking for help.

I have copied the code I made for test at the end.

# Questions
To summarise, there are three questions to be addressed:

### Q3. Did I properly set CUDA and Tensorflow-gpu on WSL2 system? Could you guide me how to set Tensorflow-gpu on WSL2?
-> To use GPU and Tensorflow in WSL2, I went through a very hard time. And possibly I did make something wrong with set up. Let me briefly explain the process I went through:
Step 1. Install CUDA v10.1
Step 2. Install CuDNN v7.6.5
Step 3. Set the environment with these commands:
$ echo 'export LD_LIBRARY_PATH=/usr/lib/cuda/lib64:$LD_LIBRARY_PATH' >> ~/.bashrc
$ echo 'export LD_LIBRARY_PATH=/usr/lib/cuda/include:$LD_LIBRARY_PATH' >> ~/.bashrc
Step 4. Install tensorflow-gpu v2.1.0 under Miniconda environment with python v3.7

I would appreciate any comment from all of you. Thanks.

### image no.1 - from the initial run, time duration = 5m 53.3s
![image](https://user-images.githubusercontent.com/49014051/157073727-0666bf17-a5ca-479d-ba74-b3d7a42adaee.png)
### image no.2 - warnings from the initial run
![image](https://user-images.githubusercontent.com/49014051/157073811-f33f43ea-daf1-45ef-b23f-85d1b6b7f608.png)

# Test code
import tensorflow as tf

mnist = tf.keras.datasets.mnist
(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10)
])

predictions = model(x_train[:1]).numpy()
tf.nn.softmax(predictions).numpy()
loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
loss_fn(y_train[:1], predictions).numpy()

model.compile(optimizer='adam',
              loss=loss_fn,
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=5)
model.evaluate(x_test,  y_test, verbose=2)

**Comment by mohantym** [comment]
Hi @jinzzasol ! Did you check instructions for [WSL ubuntu](https://docs.nvidia.com/cuda/wsl-user-guide/index.html) on Nvidia site?

**Comment by jinzzasol** [comment]
> Hi @jinzzasol ! Did you check instructions for [WSL ubuntu](https://docs.nvidia.com/cuda/wsl-user-guide/index.html) on Nvidia site too?

@mohantym Yes, I did.

**Comment by jinzzasol** [comment]
> Hi @jinzzasol ! Did you check instructions for [WSL ubuntu](https://docs.nvidia.com/cuda/wsl-user-guide/index.html) on Nvidia site too?

@mohantym I tried this again, but after I installed NVIDIA driver, now Tensorflow is using CPU, not GPU. And all the setting I made is gone now. WSL and Tensorflow don't recognise GPU anymore.

**Comment by jinzzasol** [comment]
@mohantym Now it is fixed and Tensorflow can recognise GPU. However, nothing has changed. The issues are still there.

**Comment by mohantym** [comment]
Hi @gadagashwini ! Could you please look at this issue?

**Comment by jinzzasol** [comment]
@mohantym @gadagashwini Can you please look at this issue? or can assign someone else? Thanks,

**Comment by jinzzasol** [comment]
@google-ml-butler hey

**Comment by jinzzasol** [comment]
@gadagashwini Can you look at this issue? or can assign someone else?

**Comment by mihaimaruseac** [comment]
WSL is slow in reading files. There are a lot of files that need to be loaded when `import tensorflow as tf` gets executed.

This is likely not a TF issue.

**Comment by jinzzasol** [comment]
> WSL is slow in reading files. There are a lot of files that need to be loaded when `import tensorflow as tf` gets executed.
> 
> This is likely not a TF issue.

You don't get the point. `import tensorflow as tf ` works okay and fast in WSL too. It only takes less than 0.1s.

**Comment by mihaimaruseac** [comment]
Same point applies for loading mnist data which is a bunch of image files, afaik

**Comment by jinzzasol** [comment]
> Same point applies for loading mnist data which is a bunch of image files, afaik

@mihaimaruseac Well. I tested tensorflow without using mnist data and it still very slow. I don't think the data loading takes that long. I think you still don't understand the point. When I run the code line by line, loading tensorflow and mnist data doesn't take long as you said.

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
